src/jit/codegenxarch.cpp

```diff
@@ -2689,78 +2618,23 @@ CodeGen::genStoreRegisterReturnInLclVar(GenTreePtr treeNode)
         regNumber retReg0 = REG_NA;
         regNumber retReg1 = REG_NA;
 
-        emitAttr size0 = EA_UNKNOWN;
-        emitAttr size1 = EA_UNKNOWN;
-
         unsigned __int8 offset0 = 0;
         unsigned __int8 offset1 = 0;
 
         var_types type0 = TYP_UNKNOWN;
         var_types type1 = TYP_UNKNOWN;
 
-        bool firstIntUsed = false;
-        bool firstFloatUsed = false;
-
-        genGetStructTypeSizeOffset(structDesc, &type0, &type1, &size0, &size1, &offset0, &offset1);
-
-        if (type0 != TYP_UNKNOWN)
-        {
-            if (structDesc.eightByteClassifications[0] == SystemVClassificationTypeIntegerReference ||
-                structDesc.eightByteClassifications[0] == SystemVClassificationTypeInteger)
-            {
-                retReg0 = REG_INTRET;
-                firstIntUsed = true;
-            }
-            else if (structDesc.eightByteClassifications[0] == SystemVClassificationTypeSSE)
-            {
-                retReg0 = REG_FLOATRET;
-                firstFloatUsed = true;
-            }
-            else
-            {
-                assert(false && "Invalid eightbyte type");
-            }
-        }
-
-        if (type1 != TYP_UNKNOWN)
-        {
-            if (structDesc.eightByteClassifications[1] == SystemVClassificationTypeIntegerReference ||
-                structDesc.eightByteClassifications[1] == SystemVClassificationTypeInteger)
-            {
-                if (firstIntUsed)
-                {
-                    retReg1 = REG_INTRET_1;
-                }
-                else
-                {
-                    retReg1 = REG_INTRET;
-                }
-            }
-            else if (structDesc.eightByteClassifications[1] == SystemVClassificationTypeSSE)
-            {
-                if (firstFloatUsed)
-                {
-                    retReg1 = REG_FLOATRET_1;
-                }
-                else
-                {
-                    retReg1 = REG_FLOATRET;
-                }
-            }
-            else
-            {
-                assert(false && "Invalid eightbyte type");
-            }
-        }
+        getGetStructTypeOffset(structDesc, &type0, &type1, &offset0, &offset1);
+        getReturnRegistersBasedOnTypes(structDesc, type0, type1, &retReg0, &retReg1);
 
         if (retReg0 != REG_NA)
```

Comment:
The ifs are benign. They really should be asserts. The type of the tree is struct only if this is a 2 register return. Changed to assert.